core/org.wso2.carbon.tomcat.ext/src/main/java/org/wso2/carbon/tomcat/ext/valves/CompositeValve.java

```diff
@@ -41,9 +41,12 @@ public CompositeValve() {
     @Override
     public void invoke(Request request, Response response) throws IOException, ServletException {
         try {
-            
             if (request.getContext() == null) {
-                log.error("Could not handle the request, could be due to the maxHttpHeaderSize limitation.");
+                if (request.getRequestURI() == null) {
+                    log.debug("Could not handle the request. The request URI is invalid.");
```

Comment:
```suggestion
             if (log.isDebugEnabled) {
                    log.debug("Could not handle the request. The request URI is invalid.");
             }
```

Reply:
Check the other place as well.

Reply:
addessed in ab00aabecc11f1ce72cff2eb393e7561b6747e29